lib/ZMQClient/methods/connect.js

```diff
@@ -1,56 +1,81 @@
-// eslint-disable-next-line
-const { Subscriber, SocketOptions } = require('zeromq');
+const { Subscriber } = require('zeromq');
+
+const runSocketMonitor = require('../utils/runSocketMonitor');
+const runSocketReceiver = require('../utils/runSocketReceiver');
 
 /**
  * Create a new Subscriber socket and connect to it.
  *
- * @param {SocketOptions<Subscriber>} [socketOption]
+ * @param {ConnectionOptions} [socketOption]
  * @returns {Promise<Subscriber>}
  */
-async function connect(socketOption) {
+async function connect(socketOption = {}) {
   const self = this;
+  const uri = `${this.protocol}://${this.host}:${this.port}`;
+
+  // By default, we don't set any max number of retries
+  const maxRetries = socketOption.maxRetries || -1;
+  let connectionRetries = 0;
+
   if (this.socket) {
-    await this.disconnect();
+    await this.disconnect(uri);
   }
-  const uri = `${this.protocol}://${this.host}:${this.port}`;
 
   const opts = { ...socketOption };
 
+  if (opts.maxRetries) {
+    // Subscriber opts is not extensible, we need to remove that from opts we pass
+    delete opts.maxRetries;
```

Comment:
I'm not sure how deleting it helps?

Reply:
Passing `opts` with a property named `maxRetries` to new Subscriber() would result in Subscriber to error out due to not being a valid props.

Reply:
Fixed !